src/transformers/pipelines/base.py

```diff
@@ -1082,7 +977,12 @@ def __call__(self, inputs, *args, num_workers=0, batch_size=1, **kwargs):
                 return outputs
             else:
                 return self.run_multi(inputs, preprocess_params, forward_params, postprocess_params)
-        elif Dataset is not None and isinstance(inputs, Dataset):
+        elif (
+            Dataset is not None
+            and isinstance(inputs, Dataset)
+            or isinstance(inputs, types.GeneratorType)
+            and self.framework == "pt"
```

Comment:
Would it be possible to create a boolean value out of the if statement (possibly split over several boolean values) so that this is simpler to follow?